nixio/data_frame.py

```diff
@@ -270,8 +271,12 @@ def print_table(self, row_sl=None, col_sl=None):
         print(row_form.format(" ", *cl))
         if self.units is not None:
             print(row_form.format("unit", *self.units[col_sl]))
+        if not isinstance(row_sl, Iterable):
+            ridx = list(range(len(self)))[row_sl]
+        else:
+            ridx = row_sl
         for i, row in enumerate(self._read_data(sl=row_sl)[list(cl)]):
-            print(row_form.format("Data{}".format(i), *row))
+            print(row_form.format("Data{}".format(ridx[i]), *row))
```

Comment:
Agreed.  Original (in-data) Row index is probably the most useful.  I still don't like that they're labeled `Data0, Data1, Data2,... DataN`.  An indication in the heading explaining what the number is would be more appropriate.  As it stands, `DataN` doesn't really mean much.